src/app/plugins/register.plugin.ts

```diff
@@ -16,16 +16,35 @@ export class RegisterPlugin extends Plugin {
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const request = this.container.classService.buildRequest(message.author, args);
-    if (!request) {
-      message.reply('I was unable to build your request.');
-      return;
-    }
-    try {
-      const response = await this.container.classService.register(request);
-      message.reply(response);
-    } catch (e) {
-      message.reply(e);
+    // Do a singular request if args is null or this is a category request
+    if (!args ||
+        args.length === 2 &&
+        (args[1] === ClassType.CS || args[1] === ClassType.IT)) {
+      const request = this.container.classService.buildRequest( message.author, args);
+      if (!request) {
+        message.reply('I was unable to build your request.');
+        return;
+      }
+      try {
+        const response = await this.container.classService.register(request);
+        message.reply(response);
+      } catch (e) {
+        message.reply(e);
+      }
+    } else {
+      for (let arg of args) {
+        const request = this.container.classService.buildRequest(message.author, [ arg ]);
+        if (!request) {
+          message.reply('I was unable to build your request.');
+          return;
+        }
```

Comment:
makes too many respondes

Reply:
If you mean that that it replies with `I was unable to build your request.` too many times, it should stop after the first because of the return statement immediately after. Let me know if I'm misunderstanding you.

Reply:
hmm, that also would be bad since user should get registered for any valid listed classes